lib/age_validator.rb

```diff
@@ -1,9 +1,11 @@
 class AgeValidator < ActiveModel::Validator
   DEFAULT_ADULT_AGE = 18
 
-  def initialize(prison_config)
-    @config = prison_config || (raise ArgumentError)
-    super
+  attr_accessor :adult_age
+
+  def initialize(prison_name)
+    prison = Prison.find(prison_name, true)
```

Comment:
As mentioned above, I feel that this should take the prison directly, not look it up.